src/util/sighash.rs

```diff
@@ -639,11 +640,31 @@ impl<R: Deref<Target = Transaction>> SighashCache<R> {
         Ok(Sighash::from_engine(enc))
     }
 
-    /// Encodes the legacy signing data for any flag type into a given object implementing a
-    /// [`std::io::Write`] trait. Internally calls [`Transaction::encode_signing_data_to`].
+    /// Encodes the legacy signing data from which a signature hash for a given input index with a
+    /// given sighash flag can be computed.
+    ///
+    /// To actually produce a scriptSig, this hash needs to be run through an ECDSA signer, the
+    /// [`EcdsaSighashType`] appended to the resulting sig, and a script written around this, but
+    /// this is the general (and hard) part.
+    ///
+    /// The `sighash_type` supports an arbitrary `u32` value, instead of just [`EcdsaSighashType`],
+    /// because internally 4 bytes are being hashed, even though only the lowest byte is appended to
+    /// signature in a transaction.
+    ///
+    /// # Warning
+    ///
+    /// - Does NOT attempt to support OP_CODESEPARATOR. In general this would require evaluating
+    /// `script_pubkey` to determine which separators get evaluated and which don't, which we don't
+    /// have the information to determine.
+    /// - Does NOT handle the sighash single bug (see "Return type" section)
+    ///
+    /// # Returns
+    ///
+    /// This function can handle the SIGHASH_SINGLE bug internally, so it returns [`EncodeSigningDataResult`]
```

Comment:
In 8cbc99fed84a48e02e1fd61d2ae1e630e82fab50:

`can handle` should be `can't handle`

Reply:
lolz, I think I edited that from "can't" to "cannot" then undid the edit - incorrectly. Thanks for picking it up.